yangsooplus/4주차/Week4Screen.kt

```diff
@@ -0,0 +1,271 @@
+                        onMenuSelected = {},
+                    )
+                    Week4ListMenuItem(
+                        imagePainter = ColorPainter(Color(0xFF009688)),
+                        text = "토스머니 내역 보기",
+                        onMenuSelected = {},
+                    )
+                    Week4ListMenuItem(
+                        imagePainter = ColorPainter(Color(0xFFFFC107)),
+                        text = "토스머니 충전하기",
+                        onMenuSelected = {},
+                    )
+                }
+            }
+            Week4Card {
+                Column {
+                    Text(
+                        text = "용돈기입장",
+                        fontWeight = FontWeight.Bold,
+                        fontSize = 20.sp,
+                    )
+                    Spacer(modifier = Modifier.height(24.dp))
+                    Week4ListMenuItem(
+                        imagePainter = ColorPainter(Color(0xFF20CF75)),
+                        text = "25,600원",
+                        label = "이번달 쓴 돈",
+                        onMenuSelected = {},
+                    )
+                }
+            }
+            Week4Card {
+                Text(text = "Jetpack Compose Study 파이팅")
+            }
+        }
+    }
+}
+
+@Composable
+fun Week4Card(
+    modifier: Modifier = Modifier,
+    content: @Composable () -> Unit,
+) {
+    Card(
+        modifier = modifier.fillMaxWidth().padding(16.dp),
+        colors = CardDefaults.cardColors(
+            contentColor = Color.Black,
+            containerColor = Color.White,
+        ),
+        elevation = CardDefaults.cardElevation(defaultElevation = 0.5.dp),
+        shape = RoundedCornerShape(20.dp),
+    ) {
+        Box(
+            modifier = Modifier.padding(horizontal = 24.dp, vertical = 36.dp),
+        ) {
+            content()
+        }
+    }
+}
+
+@Composable
+fun Week4ListMenuItem(
+    imagePainter: Painter,
+    text: String,
+    onMenuSelected: () -> Unit,
+    modifier: Modifier = Modifier,
+    label: String? = null,
+) {
+    Row(
+        modifier = modifier.fillMaxWidth().clickable { onMenuSelected() },
+        verticalAlignment = Alignment.CenterVertically,
+    ) {
+        Image(
+            painter = imagePainter,
+            modifier = Modifier.size(50.dp).clip(CircleShape),
+            contentDescription = null,
+        )
+        Spacer(modifier = Modifier.width(16.dp))
+        if (label == null) {
+            Text(text = text, modifier = Modifier.weight(1f), fontWeight = FontWeight.Bold)
+        } else {
+            Column(
+                modifier = Modifier.weight(1f),
+            ) {
+                Text(
+                    text = label,
+                    fontWeight = FontWeight.Light,
+                    fontSize = 14.sp,
+                    color = Color.Gray,
+                )
+                Text(text = text, fontWeight = FontWeight.Bold)
+            }
+        }
```

Comment:
```suggestion
            Column(
                modifier = Modifier.weight(1f),
            ) {
                if (label != null) {
                    Text(
                        text = label,
                        fontWeight = FontWeight.Light,
                        fontSize = 14.sp,
                        color = Color.Gray,
                    )
                }
                Text(text = text, fontWeight = FontWeight.Bold)
            }
        
```   

이렇게 Column 안에 label을 넣는건 어떠세요?
성능상에서는 매번 Column을 그리니 조금 안좋을 수는 있지만 가독성은 더 좋은거같아서요~!

Reply:
또 저는 최대한 null 값을 없애는 편이라 ... label을 non null 하게만드는건 어떻게 생각하시나요?

Reply:
제안해주신 쪽이 확실히 가독성이 더 좋네요! 
저는 보통 있다/없다를 나타낼 때는 의미를 고려해서 nullable하게 해요 
isNotEmpty()보다 null로 적혀 있으면 슥슥 읽다가도 눈에 띄더라고요

사실 언제 어느쪽을 사용해야할지는 잘 모르겠어요
매테리얼 컴포넌트에서도 선택적으로 사용하는 파라미터에 대해서 아래 두가지를 혼용하는 걸 봤었거든요

```
fun Something(
    foo: @Composable (() -> Unit)? = null
)

fun Something(
    foo: @Composable () -> Unit = {}
)
```

Reply:
아하 저는 nullable 값을 non-null 값에 대입할 때 따로 처리해주는게 불편해서 최대한 non-null하게 만들어요.
예시를 들어드리면 ...
```kotlin
val str: String? = null

fun something(parm: String) { ... }

something(str ?: "") // 엘비스 연산자를 사용해주는게 번거로움
```

---

머터리얼에서 아래 두 가지를 혼용한 이유는 아마 ... (추측) null 또는 {} 일때 특정 동작을 수행하는 코드를 짤 때 가독성을 더 좋게 하기 위함 같아요
```kotlin
fun Something(
    foo: @Composable (() -> Unit)? = null
)

fun Something(
    foo: @Composable () -> Unit = {}
)
```

### nullable
```kotlin
Something?.invoke() ?: doAnyhting()
```

### {}
```kotlin
// 이건 아마 안될거같은데 ... 의사 코드로 봐주세요 ㅎㅎ
if(Something == {}) doAnyhting() else Something()
```

kotlin에 익숙한 사람이라면 nullable이 더 가독성이 좋아보인다 생각해요
---
코틀린 스터디와는 다르지만 저희 플젝에서 코드 스타일을 맞추기 위해 다같이 한번 얘기해보는게 좋겠네요 .. ㅎㅎ

Reply:
음 다시 보니 ... 아무래도 스타일 차이 같기도 하네요 ... 하하

Reply:
TextField나 AlertDialog 같은 애들은 nullable하게 처리하고 있고, Scaffold 같은 애들은 빈 람다로 처리하고 있어서 무슨 이유가 있을까하고 들여다봐도 잘 모르겠네요 🤔 좀 더 알아봐야겠어요